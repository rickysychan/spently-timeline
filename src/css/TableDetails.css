#table-details{
    font-size: 16px;
    font-family: Karla, Helvetica, Arial, sans-serif;
}
#table-details:nth-child(even){
    background-color: var(--light-gray);    
}
.tableDetails-secondary-detail{
    text-align: right;
    padding: 5px;
}
.tableDetails-email{
    padding: 5px;
}
#table-details:hover{
    background-color: var(--teal-light);
}
