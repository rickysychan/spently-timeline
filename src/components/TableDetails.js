import React from 'react'
import PropTypes from 'prop-types'
import { convertTime } from '../helper'
import '../css/TableDetails.css'
import '../css/colors.css'

const TableDetails = ({ email, emailType, timestamp, selectCustomer, customer }) => 
            <tr className='table-details' onClick={() => selectCustomer(customer)}>
                <td className='tableDetails-email'>{email}</td>
                <td className='tableDetails-secondary-detail'>{emailType}</td>
                <td className='tableDetails-secondary-detail'>{convertTime(timestamp)}</td>
            </tr>

TableDetails.propTypes = {
    email: PropTypes.string.isRequired,
    emailType: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired
}

TableDetails.defaultProps = {
    email: 'moo@gmail.com',
    emailType: 'mooo',
    timestamp: 123
}


export default TableDetails