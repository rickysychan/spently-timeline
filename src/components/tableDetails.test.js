import React from 'react'
import { shallow } from 'enzyme'
import TableDetails from './TableDetails'

const mockSelectCustomer = jest.fn()
const mockCustomer = {sample: 'example'}
const tableDetails = shallow(<TableDetails 
                                customer={mockCustomer} 
                                selectCustomer={mockSelectCustomer}
                                />)

describe('Table', () => {
    it('should render properly', () => {
        expect(tableDetails).toMatchSnapshot()
    })
    describe('when a user clicks a row', () => {
        it('should invoke the selectCustomer method from `props`', () => {
            tableDetails.find('.table-details').simulate('click')
            expect(mockSelectCustomer).toHaveBeenCalledWith(mockCustomer)
        })
    })
})