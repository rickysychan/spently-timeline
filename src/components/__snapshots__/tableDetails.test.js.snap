// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableDetails
    customer={
        Object {
            "sample": "example",
          }
    }
    email="moo@gmail.com"
    emailType="mooo"
    selectCustomer={[Function]}
    timestamp={123}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          className="tableDetails-email"
>
          moo@gmail.com
</td>,
        <td
          className="tableDetails-secondary-detail"
>
          mooo
</td>,
        <td
          className="tableDetails-secondary-detail"
>
          Dec-31-1969 19:02:03
</td>,
      ],
      "className": "table-details",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "moo@gmail.com",
          "className": "tableDetails-email",
        },
        "ref": null,
        "rendered": "moo@gmail.com",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mooo",
          "className": "tableDetails-secondary-detail",
        },
        "ref": null,
        "rendered": "mooo",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dec-31-1969 19:02:03",
          "className": "tableDetails-secondary-detail",
        },
        "ref": null,
        "rendered": "Dec-31-1969 19:02:03",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            className="tableDetails-email"
>
            moo@gmail.com
</td>,
          <td
            className="tableDetails-secondary-detail"
>
            mooo
</td>,
          <td
            className="tableDetails-secondary-detail"
>
            Dec-31-1969 19:02:03
</td>,
        ],
        "className": "table-details",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "moo@gmail.com",
            "className": "tableDetails-email",
          },
          "ref": null,
          "rendered": "moo@gmail.com",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mooo",
            "className": "tableDetails-secondary-detail",
          },
          "ref": null,
          "rendered": "mooo",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dec-31-1969 19:02:03",
            "className": "tableDetails-secondary-detail",
          },
          "ref": null,
          "rendered": "Dec-31-1969 19:02:03",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
